05_testing_and_ci/boost_testing_precice_demo.md

```diff
@@ -0,0 +1,51 @@
+# Boost.Test in the Real World: preCICE Demo
+
+Look around preCICE in the terminal + text editor.
+
+## First Look
+
+- `cd src && ls`: preCICE is organized in several namespaces.
+- `cd math && ls`: Each namespace has a `tests` folder
+- Look at `math/tests/DifferenceTest.cpp`
+    - Imports `testing/Testing.hpp`, there we handle UTF imports
+    - Test suite for `math` namespace and test suite per file (here `math/differences.hpp`)
+    - `BOOST_CHECK` actually not recommended to use. Is used internally by `BOOST_TEST`.
+    - Powerful macro `PRECICE_TEST(1_rank)`; here it means that this test is run on one MPI rank
+        - Tests are normally run on 4 MPI ranks to test parallel implementation and to mimic different coupled solvers
+
+## Unit vs. Integration Tests
+
+- Clear separation in preCICE: integration tests only directly use API of preCICE.
```

Comment:
```suggestion
- Clear separation in preCICE: only integration tests directly use API of preCICE.
```
The suggestion is only valid if you meant to say that integration tests directly use API of preCICE, and unit tests do not.

Reply:
No, I meant indeed that integration tests use nothing but the API of preCICE. (no lower-level code)